src/main/java/mServer/crawler/sender/br/BrFilmDeserializer.java

```diff
@@ -0,0 +1,393 @@
+                .collect(Collectors.toList());
+        
+        for (int id = 0; id < bestUrls.size(); id++) {
+          final Resolution resolution = Resolution.getResolutionFromArdAudioVideoOrdinalsByProfileName(bestUrls.get(id).getVideoProfile());
+          final String url = bestUrls.get(id).getUrl();
+          
+          if (url != null && !url.isEmpty()) {
+            if (!urlMap.containsKey(resolution)) {
+              urlMap.put(resolution, url);
+            }
+          }
+        }
+      }
+    return urlMap;
+
+  }
+
+  private Optional<DatenFilm> buildFilm(final Optional<JsonObject> detailClip, final JsonObject viewer) {
+    final Optional<DatenFilm> newFilm;
+    if (detailClip.isPresent()) {
+      String description = getDescriptions(detailClip.get());
+      Map<Resolution, String> urls = getUrls(viewer);
+      
+      if(urls.containsKey(Resolution.NORMAL) && MediathekReader.urlExists(urls.get(Resolution.NORMAL))) {
+        Optional<String> subTitle = getSubtitleUrl(viewer);
+        newFilm = createFilm(detailClip.get(), description, subTitle, urls);
+        return newFilm;
+      }
+    } else {
+      printMissingDetails(JSON_ELEMENT_DETAIL_CLIP);
+    }
+    return Optional.empty();
+  }
+
+  private Optional<String> getSubtitleUrl(JsonObject viewer) {
+    if(viewer.has(JSON_ELEMENT_CLIP)) {
+      JsonObject clip = viewer.getAsJsonObject(JSON_ELEMENT_CLIP);
+      if(clip.has(JSON_ELEMENT_CAPTION_FILES)) {
+        JsonObject captionFiles = clip.getAsJsonObject(JSON_ELEMENT_CAPTION_FILES);
+        if(captionFiles.has(JSON_ELEMENT_EDGES)) {
+          JsonArray edges = captionFiles.getAsJsonArray(JSON_ELEMENT_EDGES);
+          if (edges.size() > 0) {
+            for (JsonElement edge : edges) {
+              if(edge.getAsJsonObject().has(JSON_ELEMENT_NODE)) {
+                JsonObject node = edge.getAsJsonObject().getAsJsonObject(JSON_ELEMENT_NODE);
+                if(node.has(JSON_ELEMENT_PUBLIC_LOCATION)) {
+                  return Optional.of(node.get(JSON_ELEMENT_PUBLIC_LOCATION).getAsString());
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    return Optional.empty();
+  }
+
+  private String getTheme(final JsonObject aDetailClip) {
+    
+    String theme = "";
+    
+    if (aDetailClip.has(JSON_ELEMENT_EPISODEOF)) {
+      JsonElement element = aDetailClip.get(JSON_ELEMENT_EPISODEOF);
+      if (!element.isJsonNull()) {
+        JsonObject episodeOf = aDetailClip.getAsJsonObject(JSON_ELEMENT_EPISODEOF);
+        if (episodeOf.has(JSON_ELEMENT_TITLE)) {
+          theme = episodeOf.get(JSON_ELEMENT_TITLE).getAsString();
+        }
+      }
+    }
+    
+    if (theme.isEmpty()) {
+      theme = aDetailClip.get(JSON_ELEMENT_KICKER).getAsString();
+    }
+    
+    return theme;
+  }
+  
+  private Optional<DatenFilm> createFilm(final JsonObject aDetailClip, String aDescription, Optional<String> aSubTitle, Map<Resolution, String> aUrls) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)